### PR TITLE
Database docs

### PR DESCRIPTION
**Pull request by mgax**
https://github.com/mgax/minivirt/issues/8 mentions:
> Maybe also add documentation on the image name, tags, VM ecosystem.

So here is some documentation :)

### CODE REVIEW

**Comment by mgax** [comment]
@MultisampledNight would you mind taking a look at this PR?

File: docs/Database.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+An image is an immutable snapshot of a virtual machine. It typically consists of a qcow2 disk image and a configuration file. It's identified by an `id` which is calculated based on its files using SHA256. The image files reside in a folder named `{db}/images/{id}`.
+
+### Tags
+
+An image may have tags, which are named references to the image. On disk they are implemented as symlinks to the image directory and are located at `{db}/images/{tag}`.
+
+## VMs
+
+A VM is an instance of a virtual machine. Each VM has a unique `name` and its files reside in a directory like `{db}/vms/{name}`.
```
Comment (MultisampledNight) [inline docs/Database.md]: One `VM` being an instance of a `virtual machine` sounds... recursive. Maybe you meant to write `image`, but on the other hand you wrote below that `VM`s are _usually_ derived from `image`s.

File: docs/Database.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+* `ssh-config` is a ssh configuration file for the VM.
+* `ssh-private-key` is the SSH identity key that can log into the VM.
+
+## Remotes
+
+The file `{db}/remotes.json` lists the remote repositories that are configured with the `miv remote` command.
+
+## Cache
+
+When `miv build` needs to download a file, it will save a copy in `{db}/cache`, to speed up future builds.
```
Comment (MultisampledNight) [inline docs/Database.md]: Might be worth mentioning that this cache is never cleaned up automatically, and if it's safe to delete the files there (if they're not in use).

**Review by MultisampledNight — COMMENTED** [review]
Nicely done otherwise!

File: docs/Database.md (hunk 1 above)
Comment (mgax) [reply docs/Database.md]: Right, the intention was to define the term _VM_ in the context of Minivirt. I'll make it italic.

**Pull request merged**